test5.rb

```diff
@@ -0,0 +1,136 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c])
+      c += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c], c)
+      c += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false unless yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_none?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false if yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_any?(pattern = nil)
+    c = 0
+    statement = false
+    if block_given?
+      length.times do
+        statement = true if yield(self[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      length.times do
+        statement = true if self[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_count(var = nil)
+    c = 0
+    if !var.nil?
+      to_a.length.times do |i|
+        c += 1 if to_a[i] == var
+      end
+      c
+    else
+      to_a.length
+    end
+  end
```

Comment:
### #my_count should behave as follow:
- [ ] 25. counts the number of elements yielding a true value if a block is given